lib/queuePopulator/LogReader.js

```diff
@@ -213,74 +217,69 @@ class LogReader {
                 Object.keys(publishedEntries).forEach(topic => {
                     queuedEntries[topic] = publishedEntries[topic].length;
                 });
-                const processedAll = !params
-                          || !params.maxRead
-                          || (batchState.logStats.nbLogRecordsRead
-                              < params.maxRead);
-                return done(null, {
+                const stats = {
                     readRecords: batchState.logStats.nbLogRecordsRead,
                     readEntries: batchState.logStats.nbLogEntriesRead,
+                    skippedRecords: batchState.logStats.skippedRecords,
                     queuedEntries,
-                    processedAll,
+                };
+                // Use heuristics to log when:
+                // - at least one entry is pushed to any topic
+                // - many records have been skipped (e.g. when starting up
+                //   consumption from mongo oplog)
+                // - the batch took a significant time to complete.
+                const useInfoLevel =
+                      Object.keys(stats.queuedEntries).length > 0 ||
+                      stats.skippedRecords > 1000 ||
+                      stats.elapsedMs > 10000;
```

Comment:
I think this is undefined. Is it supposed to be based off of `batchState.startTime`?

Reply:
Good catch, was a left-over from a previous way to check batch timeout, I will simply remove this check.